lnst/Recipes/ENRT/SRIOVNetnsTcRecipe.py

```diff
@@ -37,102 +37,106 @@ def add_tc_filter_rules(self, config):
-                  f"dst_mac {host2.newns.vf_eth0.hwaddr} "
-                  f"action mirred egress redirect dev {host2.vf_representor_eth0.name}")
+                  f"src_mac {host1_vf_dev.hwaddr} "
+                  f"dst_mac {host2_vf_dev.hwaddr} "
+                  f"action mirred egress redirect dev {host2_vf_rep_dev.name}")
 
         host2.run(f"tc filter add dev {host2.eth0.name} "
                   f"protocol arp ingress flower "
-                  f"src_mac {host1.newns.vf_eth0.hwaddr} "
-                  f"dst_mac {host2.newns.vf_eth0.hwaddr} "
-                  f"action mirred egress redirect dev {host2.vf_representor_eth0.name}")
+                  f"src_mac {host1_vf_dev.hwaddr} "
+                  f"dst_mac {host2_vf_dev.hwaddr} "
+                  f"action mirred egress redirect dev {host2_vf_rep_dev.name}")
 
         host2.run(f"tc filter add dev {host2.eth0.name} "
                   f"protocol arp ingress flower "
-                  f"src_mac {host1.newns.vf_eth0.hwaddr} "
+                  f"src_mac {host1_vf_dev.hwaddr} "
                   f"dst_mac FF:FF:FF:FF:FF:FF "
-                  f"action mirred egress redirect dev {host2.vf_representor_eth0.name}")
+                  f"action mirred egress redirect dev {host2_vf_rep_dev.name}")
 
-        host2.run(f"tc filter add dev {host2.vf_representor_eth0.name} "
+        host2.run(f"tc filter add dev {host2_vf_rep_dev.name} "
                   f"protocol ip ingress flower skip_sw "
-                  f"src_mac {host2.newns.vf_eth0.hwaddr} "
-                  f"dst_mac {host1.newns.vf_eth0.hwaddr} "
+                  f"src_mac {host2_vf_dev.hwaddr} "
+                  f"dst_mac {host1_vf_dev.hwaddr} "
                   f"action mirred egress redirect dev {host2.eth0.name}")
 
-        host2.run(f"tc filter add dev {host2.vf_representor_eth0.name} "
+        host2.run(f"tc filter add dev {host2_vf_rep_dev.name} "
                   f"protocol arp ingress flower "
-                  f"src_mac {host2.newns.vf_eth0.hwaddr} "
-                  f"dst_mac {host1.newns.vf_eth0.hwaddr} "
+                  f"src_mac {host2_vf_dev.hwaddr} "
+                  f"dst_mac {host1_vf_dev.hwaddr} "
                   f"action mirred egress redirect dev {host2.eth0.name}")
 
-        host2.run(f"tc filter add dev {host2.vf_representor_eth0.name} "
+        host2.run(f"tc filter add dev {host2_vf_rep_dev.name} "
                   f"protocol arp ingress flower "
-                  f"src_mac {host2.newns.vf_eth0.hwaddr} "
+                  f"src_mac {host2_vf_dev.hwaddr} "
                   f"dst_mac FF:FF:FF:FF:FF:FF "
                   f"action mirred egress redirect dev {host2.eth0.name}")
 
     @property
     def pause_frames_dev_list(self):
-        return [self.matched.host1.newns.vf_eth0, self.matched.host2.newns.vf_eth0]
+        return [self.sriov_devices[self.matched.host1].vfs[0], self.sriov_devices[self.matched.host2].vfs[0]]
```

Comment:
Thinking about this, also in the future... does it make sense to have `SRIOV devices > Host > PF/VF/VF_REPR` instead of `Host > SRIOV devices > PF/VF/VF_REPR`? 

Only possible upside I see is being able to create algorithm form Tc rules creation.

Reply:
Do you mean to change this so that we can use following?
```
        return [self.matched.host1.sriov_devices.vfs[0], self.matched.host2.sriov_devices.vfs[0]
```

Reply:
Hmm, this actually makes sense. I've pushed a change where the devices are stored directly in Host instances, see https://github.com/LNST-project/lnst/pull/346/commits/59aa3d04038f21ea8700fe0abf15861c042882f4

@Axonis @olichtne please take a look.

Reply:
yeah, this looks better.

Reply:
yeah, ack.